.circleci/config.yml

```diff
@@ -233,6 +307,68 @@ jobs:
     steps:
       - *trigger-metrics
 
+  api-diff-job:
+    parameters:
+      xcode:
+        type: string
+        default: "11.4.1"
+      device:
+        type: string
+        default: "iPhone 8 Plus"
+      iOS:
+        type: string
+        default: "13.4.1"
+      base_api_tag:
+        type: string
+        default: "v0.40.0"
+    macos:
+      xcode: << parameters.xcode >>
+    environment:
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - checkout
+      - build-api-diff-report
+      - run:
+          name: Store latest APIDiffReport
+          command: cp -R scripts/APIDiffReport ~/.
+      - run:
+          name: Checking out Base API
+          command: git checkout << parameters.base_api_tag >>
+      - restore-api-diff-cache:
+          api_diff_cache: .Environment.CIRCLE_SHA1
+      - run:
+          name: Move Api Diff
+          command: |
+            mv api_logs original_api
+      - restore-api-diff-cache-by-tag:
+          api_diff_cache: << parameters.base_api_tag >>
+      - *prepare-mapbox-file
+      - *prepare-netrc-file
+      - *update-carthage-version
+      - *restore-cache
+      - *install-dependencies
+      - *install-dependencies-12
+      - run:
+          name: Install prerequisites
+          command: if [ $(xcversion simulators | grep -cF "iOS << parameters.iOS >> Simulator (installed)") -eq 0 ]; then xcversion simulators --install="iOS << parameters.iOS >>" || true; fi
+      - *save-cache
+      - save-api-diff-cache-by-tag:
+          api_diff_cache: << parameters.base_api_tag >>
+      - run:
+          name: Restore latest APIDiffReport
+          command: |
+            rm -rf scripts/APIDiffReport
+            cp -R ~/APIDiffReport scripts/.
+      - run_api_log:
+          iOS: << parameters.iOS >>
+          device: << parameters.device >>
+      - run:
+          name: Generating MapboxCoreNavigation API Diff
+          command: cd scripts/APIDiffReport && swift run APIDiffReport diff $CIRCLE_WORKING_DIRECTORY/original_api/core_navigation_log.json $CIRCLE_WORKING_DIRECTORY/api_logs/core_navigation_log.json
+      - run:
+          name: Generating MapboxNavigation API Diff
+          command: cd scripts/APIDiffReport && swift run APIDiffReport diff $CIRCLE_WORKING_DIRECTORY/original_api/navigation_log.json $CIRCLE_WORKING_DIRECTORY/api_logs/navigation_log.json      
```

Comment:
These basically log changes to output and fail the CI. Should we Report the diff some other way?

Reply:
This is fine for now. When we work on a major version, we can make the build bot optional.